docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,492 @@
+# MeshExternalService API
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6330
+
+## Context and Problem Statement
+
+The current implementation of `ExternalService` has significant limitations and warrants a complete redesign.
+
+* Lack of support for * subdomains
+* Lack of support for thick clients (think Kafka or C* where discovery is done as part of the client).
+* Need for pluggability (for example lambda support or equivalent)
+* Working well with gateways (what if your ExternalService relies on SNI/host header to route through a gateway)
+* Applying policies with and without egress
+
+## Considered Options
+
+* Creating MeshExternalService resource and MeshPassthrough policy
+* Creating MeshExternalService
+
+## Decision Outcome
+
+Chosen option: "Creating MeshExternalService resource and MeshPassthrough policy".
+
+### Positive Consequences
+
+* Clearer structure.
+* Explicit showing what is passthough and what is external service
+* Enhanced functionalities.
+* Better policy matching capabilities.
+
+### Negative Consequences
+
+* 2 new resources
+* Temporarily increased complexity of a product until the migration is done.
+
+## Pros and Cons of the Options
+
+### Creating MeshExternalService
+
+We're introducing a new object resource `MeshExternalService`. This object will be a namespace resource and will only be creatable within the `kuma-system` namespace. It can be created within a specific zone, but in this case, it will only be accessible within the zone where it was created.
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  type: Managed # Managed
```

Comment:
I don't think we need `type` if there is only 1 type

Reply:
Might not be required field but if we would like to introduce something in the future we can consider having it

Reply:
I'd drop the type

Reply:
The only reason why I'd keep a type is if we already know there'll be other types. AFAIU that's no long the case so I'd remove type and move everthing from `managed` inside `spec`